python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -334,7 +350,16 @@ def _setParent(self, parentObj):
 
     def getCachedData(self, index):
 
-        return
+        raw_self = stripProxy(self)
+        if index > len(raw_self) or index < 0:
+            return None
+
+        if index in self._subjobIndexData.keys():
+            return self._subjobIndexData[index]
+        else:
+            return raw_self._registry.getIndexCache( stripProxy(raw_self.__getitem__(i)) )
+
+        return None
```

Comment:
I would reserve directly calling dunder methods like `__getitem__` for those times where it really matters (i.e. when implementing them to avoid recursion etc.). Other than that, doing `self[index]` is much clearer.